Makefile.base

```diff
@@ -51,6 +51,13 @@ ifeq (1, $(SUBMODULES))
                 $(patsubst $(BASE_MODULE)_%,%.c,\
                     $(filter $(BASE_MODULE)_%,$(USEMODULE)))))
 
+  # for each $(BASE_MODULE)_<name> in USEMODULE, add $(BASE_MODULE)_<name>.c to SRC
+  # unless in SUBMODULES_NO_SRC
+  SRC += $(wildcard \
+            $(filter-out $(SUBMODULES_NO_SRC),\
+                $(patsubst %,%.c,\
+                    $(filter $(BASE_MODULE)_%,$(USEMODULE)))))
```

Comment:
What is this for?

Reply:
previously `log_color.c` was not accepted (it wanted to have it named `color.c`)  this adds `log_color.c` to the accepted names.

Reply:
Why not keep the old naming pattern in line with the documentation and name the file `color.c`? It is new anyways.

Reply:
the documentation of the submodule seemed to me as if log_color.c should be an accepted name and the only reason this did not work was that it was not done the way it was intended.

my personal reason: is that my editor names the tabs with the file name (no path) and i like to find the files again and color is very generic

Reply:
Our `Makefile`'s contain already enough complexity to give me lots of nightmares. I totally agree with @miri64 that we should just use the submodule naming pattern as intended, rather than increasing the complexity of the build system even further to introduce an alternative naming pattern.

Also, adding a second naming pattern and introducing inconsistency is pretty confusing.

Reply:
@miri64 maybe #18425 is the better solution it reduces the specialty of log_* making it just 2 modules easy to create a 3rd the make files are much smaller and the overall complexity is reduced i also like the visibility of that 2 modules